core/src/main/java/org/mule/runtime/core/internal/util/rx/RetrySchedulerWrapper.java

```diff
@@ -0,0 +1,173 @@
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Wrapper for a {@code Scheduler} that retries to submit a task if a {@link RejectedExecutionException} is thrown, applying
+ * a callback each time it retries.
+ * <p>
+ * This is needed to avoid terminating a Reactor Flux when a {@link RejectedExecutionException} occurs, since
+ * `publishOn` does not support `onErrorContinue`: https://github.com/reactor/reactor-core/issues/1488
+ *
+ * @since 4.2
+ */
+public class RetrySchedulerWrapper implements Scheduler {
+
+  private final Scheduler delegate;
+  private final long retryTime;
+  private final Runnable onRetry;
+
+  public RetrySchedulerWrapper(Scheduler delegate, long retryTime, Runnable onRetry) {
+    this.delegate = delegate;
+    this.retryTime = retryTime;
+    this.onRetry = onRetry;
+  }
+
+  @Override
+  public ScheduledFuture<?> scheduleWithCronExpression(Runnable command, String cronExpression) {
+    return delegate.scheduleWithCronExpression(command, cronExpression);
+  }
+
+  @Override
+  public ScheduledFuture<?> scheduleWithCronExpression(Runnable command, String cronExpression, TimeZone timeZone) {
+    return delegate.scheduleWithCronExpression(command, cronExpression, timeZone);
+  }
+
+  @Override
+  public void stop() {
+    delegate.stop();
+  }
+
+  @Override
+  public String getName() {
+    return delegate.getName();
+  }
+
+  @Override
+  public ScheduledFuture<?> schedule(Runnable command, long delay, TimeUnit unit) {
+    return delegate.schedule(command, delay, unit);
+  }
+
+  @Override
+  public <V> ScheduledFuture<V> schedule(Callable<V> callable, long delay, TimeUnit unit) {
+    return delegate.schedule(callable, delay, unit);
+  }
+
+  @Override
+  public ScheduledFuture<?> scheduleAtFixedRate(Runnable command, long initialDelay, long period, TimeUnit unit) {
+    return delegate.scheduleAtFixedRate(command, initialDelay, period, unit);
+  }
+
+  @Override
+  public ScheduledFuture<?> scheduleWithFixedDelay(Runnable command, long initialDelay, long delay, TimeUnit unit) {
+    return delegate.scheduleWithFixedDelay(command, initialDelay, delay, unit);
+  }
+
+  @Override
+  public void shutdown() {
+    delegate.shutdown();
+  }
+
+  @Override
+  public List<Runnable> shutdownNow() {
+    return delegate.shutdownNow();
+  }
+
+  @Override
+  public boolean isShutdown() {
+    return delegate.isShutdown();
+  }
+
+  @Override
+  public boolean isTerminated() {
+    return delegate.isTerminated();
+  }
+
+  @Override
+  public boolean awaitTermination(long timeout, TimeUnit unit) throws InterruptedException {
+    return delegate.awaitTermination(timeout, unit);
+  }
+
+  private void runWithRetry(Runnable operation) {
+    while (!this.isShutdown() && !this.isTerminated()) {
+      try {
+        operation.run();
+        return;
+      } catch (RejectedExecutionException ree) {
+        onRetry.run();
+        try {
+          Thread.sleep(retryTime);
+        } catch (InterruptedException e) {
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: Either re-interrupt this method or rethrow the "InterruptedException". ([squid:S2142](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS2142))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7624&issues=AWiAldciotIXnmsUW49q&open=AWiAldciotIXnmsUW49q)

[ISSUE_KEY]: AWiAldciotIXnmsUW49q